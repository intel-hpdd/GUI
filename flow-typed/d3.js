// @flow

type MapFn<A, B> = A => B;

declare module d3 {
  declare module.exports: {
    max<A, B: number>(xs: A[], keyFn?: (A) => B): B,
    extent<A, B>(xs: A[], accessor?: MapFn<A, B>): B[],
    scale: Object,
    time: Object,
    svg: Object,
    select: Function,
    dispatch: Function
  };
}
