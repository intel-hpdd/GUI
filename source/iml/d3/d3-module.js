// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import d3 from 'd3';

export default angular.module('d3', []).value('d3', d3).name;
