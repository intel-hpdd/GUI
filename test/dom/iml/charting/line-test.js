// @flow

import Inferno from "inferno";

import Line from "../../../../source/iml/charting/line.js";
import d3 from "d3";

describe("line tests", () => {
  let svg, chartingGroup;

  beforeEach(() => {
    // $FlowFixMe: Mock for test
    Element.prototype.getTotalLength = () => 100;

    svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
    chartingGroup = document.createElementNS("http://www.w3.org/2000/svg", "g");
    svg.appendChild(chartingGroup);
  });

  afterEach(() => {
    // $FlowFixMe: Mock for test
    delete Element.prototype.getTotalLength;
  });

  it("should render as expected", () => {
    Inferno.render(
      <Line
        xScale={d3.scale
          .linear()
          .domain([0, 100])
          .range([0, 200])}
        yScale={d3.scale
          .linear()
          .domain([0, 100])
          .range([200, 0])}
        xValue={x => x}
        yValue={x => x}
        xComparator={(x, y) => x === y}
        chartingGroup={d3.select(chartingGroup).datum([0, 100])}
        color={() => "green"}
      />,
      svg
    );

    expect(svg).toMatchSnapshot();
  });
});
