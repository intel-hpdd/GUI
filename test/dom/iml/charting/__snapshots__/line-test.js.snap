// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line tests should render as expected 1`] = `
<svg>
  <g>
    <defs
      class="clip1"
    >
      <clippath
        id="clip1"
      >
        <rect
          width="200"
          height="200"
        />
      </clippath>
    </defs>
    <g
      clip-path="url(about:blank#clip1)"
      class="clipPath1"
    >
      <path
        class="line line1"
        stroke="green"
        d="M0,200L200,0"
        style="stroke-opacity: 1;"
        stroke-dasharray="100 100"
        stroke-dashoffset="100"
      />
    </g>
  </g>
</svg>
`;
